Add auth0 support

finchr opened the pull request:
Pull Request for Issue #122.

### Summary of Changes

I created an Auth0 plugin based on the Synology plugin that handles login via Auth0.

### Testing Instructions

* Auth0 configuration:
  * Create new account if needed
  * https://auth0.com/docs/quickstart/webapp/php/interactive
* Joomla Configuration
  * Install updated extension with Auth0 plugin
  * Enable plugin (Akeeba Social Login - Auth0 SSO integration)
  * Configure plugin with domain, client id, and client secret from Auth0 configuration above.

### Result before applying PR

The file plg_sociallogin_auth0.zip is not available in the install package and `Akeeba Social Login - Auth0 SSO integration` will not be available in the Plugins menu

### Result after applying PR

The file plg_sociallogin_auth0.zip is included in the install package and `Akeeba Social Login - Auth0 SSO integration` is available in the Plugins menu.

### Backwards compatibility

No, it's new plugin.

### Documentation Changes Required

It will need a new section under Social Login Plugins for auth0.

### Translation impact

No.

### Technical information

If there are technical considerations which may not be apparent in the code please explain here. For example, why you did something which is not obvious and you couldn't fully describe with code comments because it spans several files / methods.

Commit:
Working auth0 plugin!

Commit:
Add support for Auth0

nikosdion commented:
Thank you for the contribution :)

Can you please revert the files which are not relevant to the PR so we can have a cleaner merge?

Since we are at it, can you please replace the PNG logo with an SVG? I found it here https://iconduck.com/icons/27059/auth0 and you can use it, unless you have a better source.

If there are any setup instructions, I would appreciate them. If you can't write any setup instructions, no worries, it will just take me a bit longer to merge as I have to write documentation first.

Commit:
remove whitespace changes from MR

Commit:
remove whitespace changes from MR

Commit:
replace auth0.png with auth0.svg

finchr commented:
> Thank you for the contribution :)
> 
> Can you please revert the files which are not relevant to the PR so we can have a cleaner merge?
> 
> Since we are at it, can you please replace the PNG logo with an SVG? I found it here https://iconduck.com/icons/27059/auth0 and you can use it, unless you have a better source.
> 
> If there are any setup instructions, I would appreciate them. If you can't write any setup instructions, no worries, it will just take me a bit longer to merge as I have to write documentation first.

Hi @nikosdion,

Thanks for the wonderful extension! This is by far the best oauth extension out there!

- Sorry about the extra files, I removed all files with whitespace differences from this MR. 
- I added the SVG file you recommended.
- I would have taken a pass at documentation but couldn't find any in the repo. Is it maintained in another repo?

nikosdion commented:
The documentation is a wiki (https://github.com/akeeba/sociallogin/wiki) but read-only for the general public. If you paste the documentation as a comment here I can click on the Edit button on it to get the raw Markdown and put it in the wiki.

nikosdion commented:
Thank you very much for your contribution! If you have a minute to send me some documentation I will be happy to include it in the wiki :)

finchr commented:
Hi Nicholas,

I'm sorry I haven't gotten around to this yet. I just wanted to let you
know I have not forgotten!

On Thu, Oct 24, 2024 at 1:16 AM Nicholas K. Dionysopoulos <
***@***.***> wrote:

> Thank you very much for your contribution! If you have a minute to send me
> some documentation I will be happy to include it in the wiki :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/akeeba/sociallogin/pull/123#issuecomment-2434600094>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAVXM3HIWT6CYM3TFW54VY3Z5CUERAVCNFSM6AAAAABPSOMAZWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMZUGYYDAMBZGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Ray Finch
***@***.***
